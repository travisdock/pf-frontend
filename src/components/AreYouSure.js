import React from 'react';
import Popup from "reactjs-popup";

class AreYouSure extends React.Component {
    render() {
        return (
            <Popup
            open={this.props.open === "ask"}
            closeOnDocumentClick
            onClose={this.props.closeModal}
            onOpen={this.props.onOpen}
            >
            <div className="table-popup">
                <p>{this.props.message}</p>
                <button className="delete" onClick={this.props.handleDelete} >Delete</button>
                <button className="close" onClick={this.props.closeModal} >Close</button>
            </div>
            </Popup>
        )
    }
};

export default AreYouSure;