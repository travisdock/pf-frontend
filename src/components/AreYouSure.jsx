import React from 'react';
import Popup from "reactjs-popup";

class AreYouSure extends React.Component {
    render() {
        return (
            <Popup
            open={this.props.open === "ask"}
            closeOnDocumentClick
            onClose={this.props.closeModal}
            onOpen={this.props.askIfSure}
            >
            <div className="table-popup">
                <p>Are You Sure?</p>
                <button onClick={this.props.handleDelete} >Delete</button>
                <button onClick={this.props.closeModal} >Close</button>
            </div>
            </Popup>
        )
    }
};

export default AreYouSure;