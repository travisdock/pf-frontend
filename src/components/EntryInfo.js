import React from 'react';
import Popup from "reactjs-popup";

class EntryInfo extends React.Component {
    render() {
        return (
            <Popup
            open={this.props.open === "info"}
            closeOnDocumentClick
            onClose={this.props.closeModal}
            >
            <div className="table-popup">
                <p>Amount: {"$" + Number(this.props.amount).toFixed(2)}</p>
                <p>Category: {this.props.category_name}</p>
                <p>Date: {this.props.date}</p>
                <p>Notes: {this.props.notes}</p>
                <button className="delete" onClick={this.props.askIfSure}>Delete Entry</button>
                <button className="update" onClick={this.props.openUpdateModal}>Update Entry</button>
                <button className="copy" onClick={() => this.props.copyEntry(this.props)}>Copy Entry</button>
            </div>
            </Popup>
        )
    }
};

export default EntryInfo;