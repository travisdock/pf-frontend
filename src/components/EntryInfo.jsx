import React from 'react';
import Popup from "reactjs-popup";

class EntryInfo extends React.Component {
    render() {
        return (
            <Popup
            open={this.props.open === "info"}
            closeOnDocumentClick
            onClose={this.props.closeModal}
            >
            <div className="table-popup">
                <p>Amount: {"$" + Number(this.props.amount).toFixed(2)}</p>
                <p>Category: {this.props.category_name}</p>
                <p>Date: {this.props.date}</p>
                <p>Notes: {this.props.notes}</p>
                <button onClick={this.props.askIfSure}>Delete Entry</button>
                <button onClick={this.props.openUpdateModal}>Update Entry</button>
            </div>
            </Popup>
        )
    }
};

export default EntryInfo;