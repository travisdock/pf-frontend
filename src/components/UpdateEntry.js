import React from 'react';
import Popup from "reactjs-popup";

export class UpdateEntry extends React.Component {
    render() {
        return (
            <Popup
                open={this.props.open === "update"}
                closeOnDocumentClick
                onClose={this.props.closeModal}
                onOpen={this.props.openUpdateModal}
            >
                <div className="table-popup">
                <p>Update Entry</p>
                <form onSubmit={this.props.handleUpdate}>
                    <select
                        name="category_name"
                        value={this.props.category_name}
                        onChange={this.props.handleChange}
                        >
                        {this.props.current_user.categories.map(cat => <option value={cat.name} key={cat.id}>{cat.name}</option>)}
                    </select>
                    <input
                        name="date"
                        type="date"
                        value={this.props.date}
                        onChange={this.props.handleChange}
                    />
                    <input
                        name="amount"
                        placeholder="0.00"
                        value={this.props.amount}
                        onChange={this.props.handleChange}
                        onBlur={this.props.evaluateAmount}
                    />
                    <textarea
                        name="notes"
                        placeholder="Entry details..."
                        rows="5"
                        cols="55"
                        value={this.props.notes}
                        onChange={this.props.handleChange}
                    />
                    <button type="submit" className="button">
                    Submit
                    </button>
                </form>

                </div>
            </Popup>
        )
    }
};

export default UpdateEntry;
