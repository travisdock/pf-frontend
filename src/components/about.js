import React from 'react';

const About = () => (
  <div className="about">
    <h1>Sum.</h1>
    <p>Some guy once said something like "What gets measured, gets managed". For people having trouble managing their personal finances, simply measuring their income and expenses is the first step. Often just the act of knowing where your money is going brings about the motivation to make changes. Sum. is a very simple app that gives the user a lot of freedom in simply measuring. Create any number of custom categories for income and expenses. Then watch where the money goes. Once you actually see how your expenses add up it will almost be harder not to make any changes.</p>
  </div>
)

export default About;
