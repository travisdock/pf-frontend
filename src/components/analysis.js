import React from 'react';
import { connect } from 'react-redux';

import Chart from './chart';

class Analysis extends React.Component {
  state = {
    userEntries: []
  }

  componentDidMount() {
    const url = `http://localhost:3001/api/v1/month_category/${this.props.current_user.user_id}`
    fetch(url)
      .then(resp => resp.json())
      .then(resp => this.setState({userData: resp}))
  }
  render(){

    return(
      <div>{this.state.userData ?
        <div><Chart /></div>
        :
        <div>Loading...</div>
      }</div>
    )
  }
}

const mapStateToProps = state => ({
  current_user: state.current_user
});

export default connect(mapStateToProps)(Analysis);
