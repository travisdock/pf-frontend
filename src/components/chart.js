import React from 'react';
import {bb} from "billboard.js";
import { connect } from 'react-redux';

class Chart extends React.Component {

  state = {
    charts: {},
    currentChart: 0,
    stats: {},
    extraMonthInfo: {},
    load: false
  }

  componentDidMount() {
    const id = this.props.current_user.user_id

    fetch(`${process.env.REACT_APP_API}/api/v1/charts/${id}`)
      .then(resp => resp.json())
      .then(resp => {if (resp.error) {
        this.setState({error: resp.error})
      } else {
        const titles = Object.keys(resp.charts)
        const lastMonth = titles[titles.length - 1]
        this.setState({
          charts: resp.charts,
          stats: resp.stats,
          load: true,
          currentChart: resp.charts[lastMonth], currentTitle: lastMonth
        }, this.renderPieChart)
      }}
        )
  }

  selectChangePieChart = (e) => {
    const title = e.target.value
    this.setState( () => {
      return {
        currentChart: this.state.charts[title],
        currentTitle: title
      }
    }, this.renderPieChart)
  }

  renderPieChart() {
    const { currentChart, currentTitle } = this.state
    // const { currentTitle } = Object.keys(currentChart)[0]
    // debugger;
    bb.generate({
      data: {
        columns: currentChart,
        type: "pie",
        // onclick: function(d, element) { debugger }
      },
      title: {
        text: currentTitle,
        position: "top-center",
        padding: {
            top: 10,
           right: 10,
            bottom: 10,
            left: 10
        },
      },
      legend: {
        show: true,
        // contents: {
        //     bindto: "#legend",   // <ul id='legend'></ul>
        //     template: "<li style='background-color:#1f77b4'>data1</li>"
        //     or using function
        //     template: function(id, color, data) {
        //          // if you want omit some legend, return falsy value
        //          if (title !== "data1") {
        //               return "<li style='background-color:"+ color +">"+ title +"</li>";
        //          }
        //     }
        // },
        position: "bottom",  // bottom, right, inset
        equally: false,
        padding: 10,
        item: {
            // onclick: function(id) { debugger },
            // onover: function(id) { debugger },
            // onout: function(id) { debugger },
    
            // set tile's size
            tile: {
                width: 20,
                height: 15
            }
        }
      },
      bindto: "#chart"
    });
  }

  render() {
    return (
      <div className="content">
        {this.state.load ?
            <div className="chart-content">
              <select
                name="chart"
                value={this.state.currentTitle} //current chart month
                onChange={this.selectChangePieChart}
                >
                {Object.keys(this.state.charts).map(title => <option value={title} key={title} > {title} </option>) }
              </select>
              <div id="chart" />
            </div> :
            this.state.error ? <div>{this.state.error}</div> : <div>Loading...</div>
        }
      </div>
    )
  }
}
const mapStateToProps = state => ({
  current_user: state.current_user
});

export default connect(mapStateToProps)(Chart)



// This is the profit loss chart. I got rid of it because it was ugly but I may want to remake it one day so here is the code.
// renderPLChart() {
//   const { currentChart } = this.state
//   bb.generate({
//     data: {
//       x: "x",
//       columns: currentChart,
//     type: "area-spline"
//     },
//     axis: {
//       x: {
//         type: "timeseries",
//         tick: {
//           format: "%b"
//         }
//       }
//     },
//     bindto: "#chart"
//   });
// }