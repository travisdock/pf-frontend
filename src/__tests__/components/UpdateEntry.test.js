import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import { UpdateEntry } from '../../components/UpdateEntry';

const props = {
    open: 'update',
    closeModal: jest.fn(),
    openUpdateModal: jest.fn(),
    handleUpdate: jest.fn(),
    handleChange: jest.fn(),
    evaluateAmount: jest.fn(),
    category_name: 'categoryOne',
    date: '12/22/16',
    amount: '122.22',
    notes: 'This is the notes',
    current_user: {
        user_id: 1,
        categories: [
            {
                id: 1,
                name: "categoryOne"
            },
            {
                id: 2,
                name: "categoryTwo"
            }
        ]
    }
};
const closedProps = {
    open: 'false',
    current_user: {
        user_id: 1,
        categories: []
    }
}

describe('snapshots', () => {    
    it('renders an open <UpdateEntry /> snapshot', () => {
        const component = renderer.create(<UpdateEntry {...props} />);
        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('renders an closed <UpdateEntry />', () => {
        const component = renderer.create(<UpdateEntry {...closedProps} />);
        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });
});

describe('select Category methods fire appropriately', () => {
    test('handleSubmit is fired on button click', () => {
        const component = shallow(<UpdateEntry {...props} />);

        const form = component.find('form')
        form.simulate('submit', { preventDefault () {} });

        expect(props.handleUpdate).toHaveBeenCalled();
    });

    it('calls handleChange on category select change', () => {
        const component = shallow(<UpdateEntry {...props} />);
        
        const input = component.find('select').first()
        input.simulate('change', {target: { value: 'categoryTwo' } });
        
        expect(props.handleChange).toHaveBeenCalled();
    });

    it('calls handleChange on date input change', () => {
        const component = shallow(<UpdateEntry {...props} />);
        
        const input = component.find('input[type="date"]').first()
        input.simulate('change', {target: { value: '12/22/17' } });
        
        expect(props.handleChange).toHaveBeenCalled();
    });

    it('calls handleChange on amount input change', () => {
        const component = shallow(<UpdateEntry {...props} />);
        
        const input = component.find('input[name="amount"]').first()
        input.simulate('change', {target: { value: '122' } });
        
        expect(props.handleChange).toHaveBeenCalled();
    });

    it('calls evaluateAmount on amount input blur', () => {
        const component = shallow(<UpdateEntry {...props} />);
        
        const input = component.find('input[name="amount"]').first()
        input.simulate('focus');
        input.simulate('change', {target: { value: '122' } });
        input.simulate('blur');
        
        expect(props.evaluateAmount).toHaveBeenCalled();
    });

    it('calls handleChange on notes input change', () => {
        const component = shallow(<UpdateEntry {...props} />);
        
        const input = component.find('textarea').first()
        input.simulate('change', {target: { value: 'hello there' } });
        
        expect(props.handleChange).toHaveBeenCalled();
    });
});