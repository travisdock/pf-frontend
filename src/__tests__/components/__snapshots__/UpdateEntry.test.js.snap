// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshots renders an closed <UpdateEntry /> 1`] = `null`;

exports[`snapshots renders an open <UpdateEntry /> snapshot 1`] = `
Array [
  <div
    style={
      Object {
        "left": "0px",
        "position": "absolute",
        "top": "0px",
      }
    }
  />,
  <div
    className="popup-overlay "
    onClick={[Function]}
    style={
      Object {
        "background": "rgba(0, 0, 0,0.5)",
        "bottom": "0",
        "display": "flex",
        "left": "0",
        "position": "fixed",
        "right": "0",
        "top": "0",
        "zIndex": "999",
      }
    }
  >
    <div
      className="popup-content "
      onClick={[Function]}
      style={
        Object {
          "background": "rgb(255, 255, 255)",
          "border": "1px solid rgb(187, 187, 187)",
          "margin": "auto",
          "padding": "5px",
          "position": "relative",
          "width": "50%",
        }
      }
    >
      <div
        className="table-popup"
      >
        <p>
          Update Entry
        </p>
        <form
          onSubmit={[MockFunction]}
        >
          <select
            name="category_name"
            onChange={[MockFunction]}
            value="categoryOne"
          >
            <option
              value="categoryOne"
            >
              categoryOne
            </option>
            <option
              value="categoryTwo"
            >
              categoryTwo
            </option>
          </select>
          <input
            name="date"
            onChange={[MockFunction]}
            type="date"
            value="12/22/16"
          />
          <input
            name="amount"
            onBlur={[MockFunction]}
            onChange={[MockFunction]}
            placeholder="0.00"
            value="122.22"
          />
          <textarea
            cols="55"
            name="notes"
            onChange={[MockFunction]}
            placeholder="Entry details..."
            rows="5"
            value="This is the notes"
          />
          <button
            className="button"
            type="submit"
          >
            Submit
          </button>
        </form>
      </div>
    </div>
  </div>,
]
`;
